packages/fields/src/mutation/index.ts

```diff
@@ -0,0 +1,184 @@
+/**
```

Comment:
Why these actions are not top-level like any other?

Reply:
The idea is that these functions are utils and not “proper actions”

Reply:
Why is the folder called `mutation` and not `utils`?

Reply:
Looking at the code:

- `editPostWithNotices` is not used anywhere. Can we removed it?
- `deletePostWithNotices` (and the `NoticeSettings` type) is just used by the `delete-post` action. Can we inline it in that file instead?

Reply:
True! To not make this PR too big, I didn't "migrate" existent actions to these two utils, but `deletePostWithNotices` could be used:
- `permanently-delete-post`: 
https://github.com/WordPress/gutenberg/blob/400a67d954eed259514c73d69225e34b08093ef0/packages/fields/src/actions/permanently-delete-post.tsx#L16
-`trash-post`:
https://github.com/WordPress/gutenberg/blob/400a67d954eed259514c73d69225e34b08093ef0/packages/fields/src/actions/restore-post.tsx#L16

Instead `editPostWithNotices` could be used with:
- `restore-post`:
https://github.com/WordPress/gutenberg/blob/400a67d954eed259514c73d69225e34b08093ef0/packages/fields/src/actions/restore-post.tsx#L16
- `reset-post`:
https://github.com/WordPress/gutenberg/blob/400a67d954eed259514c73d69225e34b08093ef0/packages/fields/src/actions/reset-post.tsx#L179